feat(docs): add docs example dir and use include code to update current tutorials

sklppy88 opened the pull request:
I think we need to look into adding typescript code to this, but I want to do that in a follow up PR.  It currently only builds the contracts and does not run any tests, but building the contracts already provides protection against compilation errors.

I've kept changes to noir-contracts/bootstrap.sh purposefully verbose and have not done anything fancy.

We also should have codeowners on this examples directory once this is merged !

Have tested by running `./bootstrap.sh compile` in mainframe !

sklppy88 commented:
* **#17291** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17291?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17291?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

critesjosh reviewed (dismissed):
per adams feedback on slack, let's dedup these CI functions

sklppy88 commented:
> per adams feedback on slack, let's dedup these CI functions

Thanks for pinging on slack, have addressed, and have requested a review from him too

ludamad commented on noir-projects/noir-contracts/bootstrap.sh:
```diff
@@ -143,7 +163,11 @@ function compile {
   # Calculate filename because nargo...
-  contract_name=$(cat contracts/$contract_path/src/main.nr | awk '/^contract / { print $2 } /^pub contract / { print $3 }')
+  if [ -n "${DOCS_WORKING_DIR:-}" ]; then
+    contract_name=$(cat examples/$contract_path/src/main.nr | awk '/^contract / { print $2 } /^pub contract / { print $3 }')
```
Can we pass the contract folder name as an arg so we dont have conditionals here?

ludamad commented:
LGTM disabling automerge for minor feedbackl

sklppy88 committed:
feat(docs): add docs example dir and use include code to update current tutorials

I think we need to look into adding typescript code to this, but I want to do that in a follow up PR.  It currently only builds the contracts and does not run any tests, but building the contracts already provides protection against compilation errors.

I've kept changes to noir-contracts/bootstrap.sh purposefully verbose and have not done anything fancy.

We also should have codeowners on this examples directory once this is merged !

Have tested by running `./bootstrap.sh compile` in mainframe !
